iothub/service/src/Common/Amqp/ClientWebSocketTransport.cs

```diff
@@ -68,7 +68,44 @@ public override bool WriteAsync(TransportAsyncCallbackArgs args)
             return true;
         }
 
-        private async Task WriteAsyncCore(TransportAsyncCallbackArgs args)
+        public override bool ReadAsync(TransportAsyncCallbackArgs args)
+        {
+            ThrowIfNotOpen();
+
+            // Read with buffer list not supported
+            Fx.AssertAndThrow(args.Buffer != null, "must have buffer to read");
+            Fx.AssertAndThrow(args.CompletedCallback != null, "must have a valid callback");
+
+            Utils.ValidateBufferBounds(args.Buffer, args.Offset, args.Count);
+            args.Exception = null; // null out any exceptions
+
+            Task<int> taskResult = ReadAsyncCoreAsync(args);
+            if (ReadTaskDone(taskResult, args))
+            {
+                return false;
+            }
+
+            taskResult.ToAsyncResult(s_onReadComplete, args);
+            return true;
+        }
+
+        public void Dispose()
+        {
+            if (!_isDisposed)
+            {
+                if (_webSocket.State != WebSocketState.Aborted)
+                {
+                    _webSocket.Abort();
+                    _webSocket.Dispose();
```

Comment:
I don't know the context but should we not be disposing the websocket irrespective of the state when we call dispose? I can see that we only abort if the state is no Aborted.

Reply:
My only guess is that when it's aborted, it's already disposed. I could be wrong :)